### PR TITLE
Milestone 2

### PR DESCRIPTION
**Pull request by Eshy10**
## initializing the Redux actions, reducers, and store
- Created directory for components, containers, reducers, and actions.
- Added Html table in the BookList component and connect it to the store to fetch the books from the state.
-  Created a Book component to accept a prop and represented a row in the HTML table.
-  Created a Book Form component with input, select options for categories, and a submit button.
- Defined the redux actions and reducers
- Initialized the  Redux store and create an initial state object comprising titles, categories, and ids.

### CODE REVIEW

File: src/components/Book.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  <tr>
+    <td>{book.id}</td>
+    <td>{book.title}</td>
+    <td>{book.category}</td>
+  </tr>
+);
+
+Book.propTypes = {
+  // eslint-disable-next-line react/forbid-prop-types
+  book: PropTypes.object.isRequired,
```
Comment (bellom) [inline src/components/Book.js]: Using [PropTypes.shape](https://dev.to/cesareferrari/how-to-specify-the-shape-of-an-object-with-proptypes-3c56) here will allow you to set a type check that is more accurate, then you can remove the inline comment to disable prop-types.

File: src/containers/BooksList.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+const BookList = ({ books }) => {
+  const rowHead = ['ID', 'Title', 'Category'];
+  return (
+    <div>
+      <table>
+        <thead>
+          {
+         rowHead.map((heading, index) => (
+           // eslint-disable-next-line react/no-array-index-key
```
Comment (bellom) [inline src/containers/BooksList.js]: we are indirectly changing the rules by disabling of these rules, let's fix the codebase instead.

keys should not be indexes, it is bad practice, it messes up the DOM if something is removed from the collection of object.

kindly check this out 👉  [stackoverflow](https://stackoverflow.com/questions/46735483/error-do-not-use-array-index-in-keys)

File: src/containers/BooksList.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+          ))
+        }
+        </tbody>
+      </table>
+    </div>
+  );
+};
+
+BookList.propTypes = {
+  // eslint-disable-next-line react/forbid-prop-types
```
Comment (bellom) [inline src/containers/BooksList.js]: Using [PropTypes.shape](https://dev.to/cesareferrari/how-to-specify-the-shape-of-an-object-with-proptypes-3c56) here will allow you to set a type check that is more accurate, then you can remove the inline comment to disable prop-types.

File: src/reducers/index.js [hunk 4]
```diff
@@ -0,0 +1,8 @@
+import { combineReducers } from 'redux';
+import bookReducer from './books';
+
+const rootReducer = combineReducers({
+  books: bookReducer,
+});
```
Comment (bellom) [inline src/reducers/index.js:6]: ### [Optional]
I think we can shorten the code like this 👇 ;
```
const rootReducer = combineReducers({
  bookReducer,
});
```

imagine we have a lot of reducers to combine.

**Review by bellom — CHANGES_REQUESTED** [review]
## Hi Team :clap: 

### Awesome job so far!! :+1: 

#### You have done wonderfully well, you passed `almost` all the checklist for this project :+1: just a few issues left to work on............ :slightly_smiling_face: 
 
**Kindly check out my comments below;**

### ♻️ Kindly re-submit for review after making all those changes :slightly_smiling_face: Thumps -up await you.....

**Best,**

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

File: src/components/Book.js (hunk 1 above)
Comment (Eshy10) [reply src/components/Book.js]: okay

File: src/containers/BooksList.js (hunk 2 above)
Comment (Eshy10) [reply src/containers/BooksList.js]: fixed. thanks

**Review by bellom — APPROVED** [review]
## Hi Team :clap: 

**Great work making all the necessary corrections from the previous review.**

## Congratulation :+1:  
_Go ahead and merge to dev and don't forget to submit the project completion form.
Good luck in the next milestone :slightly_smiling_face:_ 

**Best,**

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

**Pull request merged**